pkg/api/payer/service.go

```diff
@@ -68,10 +70,66 @@ func NewPayerApiService(
 		payerPrivateKey:     payerPrivateKey,
 		blockchainPublisher: blockchainPublisher,
 		nodeCursorTracker:   NewNodeCursorTracker(ctx, log, metadataClient),
-		nodeSelector:        &StableHashingNodeSelectorAlgorithm{reg: registry},
+		nodeSelector:        &StableHashingNodeSelectorAlgorithm{reg: nodeRegistry},
+		nodeRegistry:        nodeRegistry,
 	}, nil
 }
 
+// GetReaderNode returns a reader node and a list of backup nodes.
+// For now, the reader node is chosen randomly from the list of nodes.
+// In the future, different algorithms can be implemented and selected in the request.
+func (s *Service) GetReaderNode(
+	ctx context.Context,
+	req *payer_api.GetReaderNodeRequest,
+) (*payer_api.GetReaderNodeResponse, error) {
+	var (
+		start       = time.Now()
+		queryStatus = "success"
+		nodes       []registry.Node
+	)
+
+	defer func() {
+		metrics.EmitPayerGetReaderNodeDuration(time.Since(start).Seconds(), queryStatus)
+		metrics.EmitPayerGetReaderNodeAvailableNodes(len(nodes))
+	}()
+
+	var err error
+	nodes, err = s.nodeRegistry.GetNodes()
```

Comment:
you don't need to declare either nodes nor error. You can just use `nodes, err := s.nodeRegistry.GetNodes()`

Reply:
up to you :) Your version is probably cleaner